src/job/stagingJobHandler.ts

```diff
@@ -75,10 +77,30 @@ export class StagingJobHandler extends JobHandler {
     }
   }
   async deploy(): Promise<CommandExecutorResponse> {
+    let resp;
     try {
-      const resp = await this.deployGeneric();
+      if (process.env.FEATURE_FLAG_MONOREPO_PATH === 'true' && this.currJob.payload.repoName === MONOREPO_NAME) {
+        const repo_info = await this._docsetsRepo.getRepo(
+          this.currJob.payload.repoName,
+          this.currJob.payload.directory
+        );
+        if (!repo_info) {
+          const errorMessage = `
+            Docset Repo data not found in Atlas for repoName: ${this.currJob.payload.repoName}\n
+            ${this.currJob.payload.directory ? `directory: ${this.currJob.payload.directory}\n` : ''}
+            project: ${this.currJob.payload.project}
+          `;
+          throw Error(errorMessage);
+        }
+
+        const { bucket, url } = await this.getEnvironmentVariables();
+        resp = await nextGenStage({ job: this.currJob, bucket, url });
```

Comment:
NIT: It appears that we no longer throw an error in `nextGenStage`, we will never log to `stdErr` [here](https://github.com/mongodb/docs-worker-pool/blob/ffe258c96da8516743c5844fcadd209da194715e/src/job/stagingJobHandler.ts#L110C28-L110C28). We can either add throw statements back to `nextGenStage` or check the result of the the response status to see if it has failed.